From 168c4292036c75910a3a5e588d34fbda68174d3d Mon Sep 17 00:00:00 2001
From: Simon Dinkin <simonx.dinkin@intel.com>
Date: Wed, 26 Oct 2016 17:40:58 +0300
Subject: [PATCH] mbo: configure assoc disallow to reject associations from all
 STA

in the original implementation, the assoc disallow feature was to
reject associaion only for mbo STA. fix it to reject association
from all STA.

Signed-off-by: Simon Dinkin <simonx.dinkin@intel.com>
---
 src/ap/drv_callbacks.c | 3 +--
 src/ap/ieee802_11.c    | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/ap/drv_callbacks.c b/src/ap/drv_callbacks.c
index 469e185..f026dc6 100644
--- a/src/ap/drv_callbacks.c
+++ b/src/ap/drv_callbacks.c
@@ -368,8 +368,7 @@ int hostapd_notif_assoc(struct hostapd_data *hapd, const u8 *addr,
 #ifdef CONFIG_MBO
 	mbo_ap_check_sta_assoc(hapd, sta, &elems);
 
-	if (hapd->conf->mbo_enabled && hapd->mbo_bss_assoc_disallow &&
-		sta->mbo_supported) {
+	if (hapd->conf->mbo_enabled && hapd->mbo_bss_assoc_disallow) {
 		wpa_printf(MSG_INFO, "MBO: AP configure to reject associations");
 		return WLAN_STATUS_AP_UNABLE_TO_HANDLE_NEW_STA;
 	}
diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index a6b4edd..cacfc10 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -1848,8 +1848,7 @@ static u16 check_assoc_ies(struct hostapd_data *hapd, struct sta_info *sta,
 #ifdef CONFIG_MBO
 	mbo_ap_check_sta_assoc(hapd, sta, &elems);
 
-	if (hapd->conf->mbo_enabled && hapd->mbo_bss_assoc_disallow &&
-		sta->mbo_supported) {
+	if (hapd->conf->mbo_enabled && hapd->mbo_bss_assoc_disallow) {
 		wpa_printf(MSG_INFO, "MBO: AP configure to reject associations");
 		return WLAN_STATUS_AP_UNABLE_TO_HANDLE_NEW_STA;
 	}
-- 
2.10.1

